core/src/main/java/org/apache/iceberg/util/JsonUtil.java

```diff
@@ -129,6 +130,42 @@ public static String getStringOrNull(String property, JsonNode node) {
     return builder.build();
   }
 
+  public static Map<String, String> getStringMapOrNull(String property, JsonNode node) {
+    if (!node.has(property)) {
+      return null;
+    }
+
+    JsonNode pNode = node.get(property);
+    if (pNode == null || pNode.isNull()) {
+      return null;
+    }
+
+    Preconditions.checkArgument(pNode.isObject(),
+        "Cannot parse %s from non-object value: %s", property, pNode);
```

Comment:
I would move this check into the `buildStringMap...` method, since that method requires the node is an object.

Reply:
The idea was to leave `buildStringMap...` with no precodition checks, and then have the individual `getStringMap` and `getStringMapOrNull` call out to that function after doing their preconditions check.

I left the current implementation of `getStringMap` as is to minimize the diff, but I can either:

Reply:
After some changes, `getStringMapOrNull` wound up not being needed, and thus this function wasn't needed either.